### DISCUSSION TITLE
Unifi Guest Wifi image QRCODE

### DISCUSSION THREAD

**Discussion by luckyzor** [Blackymas/NSPanel_HA_Blueprint#2440] [Q&A]
Hello,

I want to use HA to change randomly the Guest wifi pw every month.
this is already possible, but the new password is in the QR Code generated by HA eg.: image.azkaban_guest_qr_code

Can this be displayed in the NSPanel?

Thank you for your help,

**Comment by edwardtfn** [comment]
Not yet, but we will implement a dynamic QR code on the next release, so you will be able to get any text from a helper or a text sensor, for example, which would make it possible

**Reply by luckyzor** [reply]
unfortunately, text is not possible with the unifi integration.
because of security reasons, there is no entity with the password.
we only have a image.* . here the integration page:
![image](https://github.com/user-attachments/assets/86419298-8f3c-4540-ab7e-688ed094a90d)

**Reply by edwardtfn** [reply]
Well, in this case, it's not possible, as we cannot handle images yet.

By the way, I don't understand the security thing, as the image contains a plain text in a machine readable format, so this is not safer than a plain text sensor. Or am I missing something?

**Reply by luckyzor** [reply]
You can find more information here:
https://github.com/home-assistant/core/pull/114419

Do you think it would be possible in the future?

**Reply by edwardtfn** [reply]
Yeah, it may be possible in the future.
Right now I'm working on improving the baud rate between the ESP32 and the Nextion display in a reliable way with a dreaming goal of enabling image display, which certainly could be used to display a QR code image provided by that integration.

**Reply by luckyzor** [reply]
should I create an issue for an improvement?

**Reply by edwardtfn** [reply]
Yes, please.
